pkg/controller/iampolicy/iampolicy_controller.go

```diff
@@ -120,9 +121,11 @@ func (r *ReconcileIAMPolicy) Reconcile(request reconcile.Request) (reconcile.Res
 
 		iamPolicyId = *createOutput.Policy.PolicyId
 		iamPolicyArn := *createOutput.Policy.Arn
+		iamPolicyAwsName := *createOutput.Policy.PolicyName
```

Comment:
You need to validate that Policy is not nil before dereferencing PolicyName

Reply:
Wait, is that true? I'm willing to bet it is. If Policy is a pointer type then you do.

Reply:
I'll add in nil checks to all of the controllers when dereferencing the createOutput struct. (Or similar situations)